lib/buffer.js

```diff
@@ -590,11 +589,8 @@ Buffer.prototype.toString = function(encoding, start, end) {
 
     if (end <= start)
```

Comment:
This is much clearer now, had to double check `undefined <= undefined` returns false due to number hint.